### PR TITLE
Highlight active link in nav bar

### PR DESCRIPTION
**Pull request by laurakwhit**
## Description & motivation 💭 
Indicates the active navigation link in the sidebar with the same style as on hover.

### Screenshots (if applicable) 📸 

|  | |
|-|-|
|Screenshot 2024-09-12 at 11 10 13 AM|Screenshot 2024-09-12 at 11 10 36 AM|

### How was this tested 👻 

- [x] Manual testing
- [ ] E2E tests added
- [ ] Unit tests added

### Issue(s) closed 
`DT-2429`

### CODE REVIEW

File: src/lib/holocene/navigation/navigation-item.svelte [hunk 1]
```diff
@@ -29,7 +32,8 @@
     aria-disabled={disabled}
     class:disabled
     tabindex={disabled ? -1 : 0}
-    class="mb-1 flex items-center whitespace-nowrap rounded-lg p-1 pl-2 text-sm font-medium hover:bg-white hover:text-black group-[.surface-primary]:hover:bg-black group-[.surface-primary]:hover:text-white group-[.surface-primary]:dark:hover:bg-white group-[.surface-primary]:dark:hover:text-black"
+    class:active={isActive($page.url.href)}
```
Comment (Alex-Tideman) [inline src/lib/holocene/navigation/navigation-item.svelte:35]: Get a lot of errors for this lil guy

File: src/routes/(app)/+layout.svelte [hunk 2]
```diff
@@ -56,45 +56,67 @@
...
     inProgressBatch: boolean,
   ): NavLinkListItem[] => {
+    const workflowsRoute = routeForWorkflows({ namespace });
+    const schedulesRoute = routeForSchedules({ namespace });
+    const batchOperationsRoute = routeForBatchOperations({ namespace });
+    const archivalRoute = routeForArchivalWorkfows({ namespace });
+    const namespacesRoute = routeForNamespaces();
+    const nexusRoute = routeForNexus();
+    const historyImportRoute = routeForEventHistoryImport();
+
```
Comment (Alex-Tideman) [inline src/routes/(app)/+layout.svelte:66]: These might need to all be reactive

Comment (Alex-Tideman) [reply src/routes/(app)/+layout.svelte:66]: Oh wait no it's in a function

File: src/lib/holocene/navigation/navigation-item.svelte (hunk 1 above)
Comment (Alex-Tideman) [reply src/lib/holocene/navigation/navigation-item.svelte:35]: Whenever you change namespace in namespace switcher it errors and crashes UI

Comment (laurakwhit) [reply src/lib/holocene/navigation/navigation-item.svelte]: Added a [fix](https://github.com/temporalio/ui/pull/2324/commits/79b803a6370752e13cf6ac819584aa5d7cb3edb3). Looks like `isActive` was still `undefined` despite setting `() => false` as the default initial value 🤔

**Review by Alex-Tideman — APPROVED** [review]

**Pull request merged**